src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -1319,6 +1311,111 @@ nns_edge_create_handle (const char *id, nns_edge_connect_type_e connect_type,
     goto error;
   }
 
+  *edge_h = eh;
+  return NNS_EDGE_ERROR_NONE;
+
+error:
+  nns_edge_release_handle (eh);
+  return ret;
+}
+
+/**
+ * @brief Create edge custom handle.
+ */
+int
+nns_edge_custom_create_handle (const char *id, const char *lib_path,
+    nns_edge_node_type_e node_type, nns_edge_h * edge_h)
+{
+  int ret = NNS_EDGE_ERROR_NONE;
+  nns_edge_handle_s *eh;
+  void *custom_handle;
+  NnsEdgeCustomDef *custom_h;
+
+  if (node_type < 0 || node_type >= NNS_EDGE_NODE_TYPE_UNKNOWN) {
+    nns_edge_loge ("Invalid param, set exact node type.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!STR_IS_VALID (lib_path)) {
+    nns_edge_loge ("Invalid param, given custom lib path is invalid.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!edge_h) {
+    nns_edge_loge ("Invalid param, edge_h should not be null.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  ret = _nns_edge_create_handle (id, node_type, edge_h);
+  if (ret != NNS_EDGE_ERROR_NONE) {
+    nns_edge_loge ("Failed to create edge handle.");
+    return ret;
+  }
+  eh = (nns_edge_handle_s *) * edge_h;
+  eh->connect_type = NNS_EDGE_CONNECT_TYPE_CUSTOM;
+
+  custom_handle = dlopen (lib_path, RTLD_LAZY);
+  if (custom_handle) {
+    void *(*getCustomHandle) () =
+        (void *(*)()) dlsym (custom_handle, "nns_edge_custom_get_handle");
+    if (!getCustomHandle) {
+      nns_edge_loge ("Failed to find nns_edge_custom_get_handle: %s",
+          dlerror ());
+      return NNS_EDGE_ERROR_UNKNOWN;
+    }
+    eh->custom_h = getCustomHandle ();
+  } else {
+    nns_edge_loge ("Failed to open custom handle: %s]\n", dlerror ());
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;;
+  }
+
```

Comment:
The code is perfect!! :+1: 
I think extracting the part that loads the custom library into a function would be good. How do you feel? :smile: 
```
    ret = load_custom_library(lib_path, &custom_handle, &custom_h);
    if (ret != 0) {
        return NNS_EDGE_ERROR_INVALID_PARAMETER;
    }
```
```
int load_custom_library(const char *lib_path, void **handle, NnsEdgeCustomDef **custom_h) {
    *handle = dlopen(lib_path, RTLD_LAZY);
    if (*handle == NULL) {
        fprintf(stderr, "Failed to open custom handle: %s\n", dlerror());
        return -1;
    }

    void *(*getCustomHandle)() = (void *(*)()) dlsym(*handle, "nns_edge_custom_get_handle");
    if (!getCustomHandle) {
        fprintf(stderr, "Failed to find nns_edge_custom_get_handle: %s\n", dlerror());
        dlclose(*handle);
        *handle = NULL;
        return -1;
    }
```
    *custom_h = (NnsEdgeCustomDef *) getCustomHandle();
    return 0;
}

void unload_custom_library(void *handle) {
    if (handle != NULL) {
        dlclose(handle);
    }
}
```

Reply:
I think you've implemented it well. Thank you. 👍